src/api/app/models/bs_request.rb

```diff
@@ -806,41 +805,38 @@ def change_review_state(new_review_state, opts = {})
         p[:description_extension] = superseded_by.to_s
         save!
         history.create(p)
-      elsif go_new_state # either no open reviews anymore or going back to review
-        if go_new_state == :new
-          history = HistoryElement::RequestAllReviewsApproved
-          # if it would go to new, we need to check if all groups agree
-          bs_request_action_groups.each do |g|
-            if g.find_review_state_of_group == :review
-              go_new_state = nil
-              history = nil
-            end
-          end
-          # if all groups agreed, we can set all now to new
-          bs_request_action_groups.each(&:set_group_to_new) if go_new_state
-        elsif go_new_state == :review
-          bs_request_action_groups.each(&:set_group_to_review)
-        elsif go_new_state == :declined
-          history = HistoryElement::RequestDeclined
-        else
-          raise "Unhandled state #{go_new_state} for history"
-        end
-        self.state = go_new_state if go_new_state
+        return
+      end
 
-        self.commenter = User.current.login
-        self.comment = opts[:comment]
-        self.comment = 'All reviewers accepted request' if go_new_state == :accepted
+      new_request_state = calculate_state_from_reviews
+      return if new_request_state == old_request_state
+      self.comment = opts[:comment]
+      self.state = new_request_state
+      self.commenter = User.current.login
+      if new_request_state == :new
+        history = HistoryElement::RequestAllReviewsApproved
+        # if all groups agreed, we can set all now to new
+        bs_request_action_groups.each(&:set_group_to_new)
+        self.comment = 'All reviewers accepted request'
+        save!
+        Event::RequestReviewsDone.create(notify_parameters)
+        # pre-approved requests can be processed
+        BsRequestAutoAcceptJob.perform_later(id) if approver
+      elsif new_request_state == :review
+        bs_request_action_groups.each(&:set_group_to_review)
+        save!
+      elsif new_request_state == :declined
+        history = HistoryElement::RequestDeclined
+        save!
+      else
+        raise "Unhandled state #{go_new_state} for history"
       end
-      save!
       history.create(p) if history
-
-      # pre-approved requests can be processed
-      BsRequestAutoAcceptJob.perform_later(id) if go_new_state == :new && approver
     end
   end
 
   def check_if_valid_review!(opts)
-    return unless !opts[:by_user] && !opts[:by_group] && !opts[:by_project]
+    return if opts[:by_user] || opts[:by_group] || opts[:by_project]
```

Comment:
@Ana06 this is btw ssomething for your slides, because what happend rubocop rewrote

```
-    if !opts[:by_user] && !opts[:by_group] && !opts[:by_project]
-      raise InvalidReview.new
-    end
+    return unless !opts[:by_user] && !opts[:by_group] && !opts[:by_project]
+    raise InvalidReview.new
```

Turning rather logic code into a double negation - and I don't dislike double negations :)